scipy/signal/filter_design.py

```diff
@@ -3053,18 +3057,209 @@ def ellipap(N, rp, rs):
     return z, p, k
 
 
-def besselap(N):
-    """Return (z,p,k) for analog prototype of an Nth order Bessel filter.
+def _bessel_poly(n, reverse=False):
+    """
+    Return the coefficients of Bessel polynomial of degree `n`
+
+    If `reverse` is true, a reverse Bessel polynomial is output.
+
+    Output is a list of coefficients:
+    [1]                   = 1
+    [1,  1]               = 1*s   +  1
+    [1,  3,  3]           = 1*s^2 +  3*s   +  3
+    [1,  6, 15, 15]       = 1*s^3 +  6*s^2 + 15*s   +  15
+    [1, 10, 45, 105, 105] = 1*s^4 + 10*s^3 + 45*s^2 + 105*s + 105
+    etc.
+
+    Output is a Python list of arbitrary precision long ints, so n is only
+    limited by your hardware's memory.
+
+    Sequence is http://oeis.org/A001498 , and output can be confirmed to
+    match http://oeis.org/A001498/b001498.txt :
+
+    i = 0
+    for n in range(51):
+        for x in bessel_poly(n, reverse=True):
+            print i, x
+            i += 1
+    """
+    if abs(int(n)) != n:
+        raise ValueError("Polynomial order must be a nonnegative integer")
+    else:
+        n = int(n)  # np.int32 doesn't work, for instance
+
+    out = []
+    for k in range(n + 1):
+        num = factorial(2*n - k, exact=True)
+        den = 2**(n - k) * (factorial(k, exact=True) *
+                            factorial(n - k, exact=True))
+        out.append(num // den)
+
+    if reverse:
+        return out[::-1]
+    else:
+        return out
+
+
+def _campos_zeros(n):
+    """
+    Return approximate zero locations of Bessel polynomials y_n(x) for order
+    `n` using polynomial fit (Campos-Calderon 2011)
+    """
+    if n == 1:
+        return asarray([-1+0j])
+
+    s = npp_polyval(n, [0, 0, 2, 0, -3, 1])
+    b3 = npp_polyval(n, [16, -8]) / s
+    b2 = npp_polyval(n, [-24, -12, 12]) / s
+    b1 = npp_polyval(n, [8, 24, -12, -2]) / s
+    b0 = npp_polyval(n, [0, -6, 0, 5, -1]) / s
+
+    r = npp_polyval(n, [0, 0, 2, 1])
+    a1 = npp_polyval(n, [-6, -6]) / r
+    a2 = 6 / r
+
+    k = np.arange(1, n+1)
+    x = npp_polyval(k, [0, a1, a2])
+    y = npp_polyval(k, [b0, b1, b2, b3])
 
-    The filter is normalized such that the filter asymptotes are the same as
-    a Butterworth filter of the same order with an angular (e.g. rad/s)
-    cutoff frequency of 1.
+    return x + 1j*y
+
+
+def _aberth(f, fp, x0, tol=1e-15, maxiter=50):
+    """
+    Given a function `f`, its first derivative `fp`, and a set of initial
+    guesses `x0`, simultaneously find the roots of the polynomial using the
+    Aberth-Ehrlich method.
+
+    ``len(x0)`` should equal the number of roots of `f`.
+
+    (This is not a complete implementation of Bini's algorithm.)
```

Comment:
Also here a full reference would be useful (did find this one though: http://www.researchgate.net/profile/Dario_Bini/publication/225654837_Numerical_computation_of_polynomial_zeros_by_means_of_Aberth's_method/links/00b49526d73624b29c000000.pdf)